sklearn/_isotonic.pyx

```diff
@@ -62,9 +60,9 @@ def _inplace_contiguous_isotonic_regression(floating[::1] y, floating[::1] w):
             i = k
 
 
-def _make_unique(cnp.ndarray[dtype=floating] X,
-                 cnp.ndarray[dtype=floating] y,
-                 cnp.ndarray[dtype=floating] sample_weights):
+def _make_unique(const floating[::1] X,
+                 const floating[::1] y,
+                 const floating[::1] sample_weights):
```

Comment:
I think X is two dimensional here.

```suggestion
def _make_unique(
    const floating[:, ::1] X,
    const floating[::1] y,
    const floating[::1] sample_weights,
):
```

Note that the number of dimensions was not provided previously.

Reply:
@jjerphan Thomas also referred to this modification, prompting me to revise the code.